mapper/tedge_mapper/src/c8y_mapper.rs

```diff
@@ -1,29 +1,44 @@
+use crate::c8y_converter::CumulocityConverter;
+use crate::component::TEdgeComponent;
 use crate::mapper::*;
-use mqtt_client::Topic;
+use async_trait::async_trait;
+use tedge_config::TEdgeConfig;
+use tracing::{debug_span, Instrument};
 
-#[derive(Debug)]
-pub struct CumulocityMapperConfig {
-    pub in_topic: Topic,
-    pub out_topic: Topic,
-    pub errors_topic: Topic,
-}
+const CUMULOCITY_MAPPER_NAME: &str = "tedge-mapper-c8y";
+
+pub struct CumulocityMapper {}
 
-impl Default for CumulocityMapperConfig {
-    fn default() -> Self {
-        Self {
-            in_topic: Topic::new("tedge/measurements").unwrap(),
-            out_topic: Topic::new("c8y/measurement/measurements/create").unwrap(),
-            errors_topic: Topic::new("tedge/errors").unwrap(),
-        }
+impl CumulocityMapper {
+    pub fn new() -> CumulocityMapper {
+        CumulocityMapper {}
     }
 }
 
-impl From<CumulocityMapperConfig> for MapperConfig {
-    fn from(value: CumulocityMapperConfig) -> Self {
-        MapperConfig {
-            in_topic: value.in_topic,
-            out_topic: value.out_topic,
-            errors_topic: value.errors_topic,
-        }
+#[async_trait]
+impl TEdgeComponent for CumulocityMapper {
+    async fn start(&self, tedge_config: TEdgeConfig) -> Result<(), anyhow::Error> {
+        let mapper_config = MapperConfig {
+            in_topic: make_valid_topic_or_panic("tedge/measurements"),
+            out_topic: make_valid_topic_or_panic("c8y/measurement/measurements/create"),
+            errors_topic: make_valid_topic_or_panic("tedge/errors"),
```

Comment:
The code should not panic (is this infallible?), we tend to use `thiserror::Error` to handle errors.

Reply:
The previous code called `unwrap()`, so this is an improvement on that. It won't panic because the topic names are valid - hence the name of the function.